Golem/ActivityLoop.cs

```diff
@@ -124,11 +103,71 @@ await foreach (string json in EnumerateMessages(reader).WithCancellation(_token)
         finally
         {
             _logger.LogInformation("Activity monitoring loop closed. Current job clenup");
-            setAllJobsFinished();
+            jobs.SetAllJobsFinished();
             setCurrentJob(null);
         }
     }
 
+    private Task<List<Job>> updateJobs(
+        List<ActivityState> activityStates,
+        IJobsUpdater jobs
+    ) {
+        return Task.FromResult(activityStates
+            .Select(async state => await updateJob(state, jobs))
+                .Select(task => task.Result)
+                .Where(job => job != null)
+                .Cast<Job>()
+                .ToList());
+    }
+
+    /// <param name="activityState"></param>
+    /// <param name="setCurrentJob"></param>
+    /// <param name="jobs"></param>
+    /// <returns>optional current job</returns>
+    private async Task<Job?> updateJob(
+        ActivityState activityState,
+        IJobsUpdater jobs
+    )
+    {
+        if (activityState.AgreementId == null)
+            return null;
+        var agreement = await GetAgreement(activityState.AgreementId);
+        if (agreement?.Demand?.RequestorId == null)
+        {
+            _logger.LogDebug($"No agreement for activity: {activityState.Id} (agreement: {activityState.AgreementId})");
+            jobs.SetAllJobsFinished();
```

Comment:
Because data was invalid, so I assumed it would be better to just clean all jobs.
I removed setting them all to finished. Now it just returns `null`.